srcdeps-core/src/main/java/org/l2x6/srcdeps/core/config/Configuration.java

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright 2015-2016 Maven Source Dependencies
+ * Plugin contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.l2x6.srcdeps.core.config;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.l2x6.srcdeps.core.BuildRequest.Verbosity;
+
+/**
+ * A configuration suitable for a build system wanting to handle source dependencies.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public class Configuration {
+    public static String CONFIG_MODEL_VERSION = "1.0";
+
+    public static class Builder {
+
+        private Set<String> forwardProperties = new LinkedHashSet<>();
+        private BuilderIo builderIo;
+        private List<ScmRepository> repositories = new ArrayList<>();
+        private boolean skip = false;
```

Comment:
Are the options described somewhere? Similar concerns as from the readme: I could guess what they are meant to be, but I can't be sure.